src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,238 @@
+        Node<T> current = head;
+        return getNodeByIndex(index, current);
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> setNode = new Node<>(null, value, null);
+        Node<T> currentAfter;
+        Node<T> removed;
+        Node<T> current = head;
+        if (index == 0) {
+            removed = head;
+            currentAfter = head.next;
+            setNode.next = currentAfter;
+            currentAfter.prev = setNode;
+            setNode.prev = null;
+            head = setNode;
+        } else if (index == 1) {
+            removed = head.next;
+            currentAfter = removed.next;
+            setNode.prev = head;
+            setNode.next = currentAfter;
+            head.next = setNode;
+            currentAfter.prev = setNode;
+        } else {
+            removed = (Node<T>) getNodeByIndex(index, current);
+        }
+        return removed.value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removed;
+        Node<T> current = head;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index wrong");
+        }
+        if (index == 0) {
+            if (size == 1) {
+                head = null;
+            } else {
+                head = head.next;
+                head.prev = null;
+            }
+        }
+        if (index > 0 && index < size - 1) {
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            removed = current.next;
+            unlink(removed);
+        }
+        if (index == size - 1 && index != 0) {
+            current = tail.prev;
+            current.next = null;
+            tail = current;
+        }
+        size--;
+        return current.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> current = head;
+        Node<T> removed;
+        int counter = 0;
+        for (int i = 0; i < size; i++) {
+            if (size == 1) {
+                head = null;
+                size--;
+                counter++;
+                break;
+            }
+            if (Objects.equals(object, head.value)) {
+                head = head.next;
+                head.prev = null;
+                size--;
+                counter++;
+                break;
+            }
+            if (Objects.equals(object, current.value)) {
+                removed = current.next;
+                unlink(removed);
+                size--;
+                counter++;
+                break;
+            }
+            current = current.next;
+        }
+        return (counter != 0);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return (size == 0);
+    }
+
+    private T getNodeByIndex(int index, Node<T> current) {
```

Comment:
Нам не потрібно передавати в цей метод ноду. Лише індекс
```suggestion
    private T getNodeByIndex(int index) {
```